### ISSUE TITLE
Channels will no longer play with newest nightly

### ISSUE THREAD

**Issue by DiR3Wolf** [eracknaphobia/service.psvue.epg#33]
@eracknaphobia @snoopyjoe I noticed there has been a lot of changes recently in the latest Kodi nightly(June29) so I decided to upgrade from a month old nightly.  EPG data and all is working as usual. PS Vue addon itself works perfect. However, channels won't play from PVR (I tried inputstream enabled/disabled). Can you test latest nightly and see if you can replicate issue?

Thanks!

**Comment by snoopyjoe** [comment]
Yeah my nightly build is from 6/20 and it has the same issue. Kodi default inputstream can't demux the Sony encrypted m3u8 streams. Try using the epg service from my Test branch. It eliminates the webserver and uses the plugin:// url instead of http:// in playlist file. Playlist in Kodi v18 can play links directly with plugin://, but the Kodi devs couldn't confirm if it works in v17.

Remember to delete your old playlist.m3u file so the new one can be created.

**Comment by DiR3Wolf** [comment]
Yes, I discovered your Test branch after I realized you had the same issue. I saw you were able to make it work using the plugin links. Back when this service was using cherrypi, I tried to do something similar to what you have done, which worked as far as playing the streams, but the problem was that the channel/show information was lost in playback/redirection.  Not sure what you did different (or if further changes in Kodi fixed this issue) but I'm grateful. This is how I wanted this service to work from the very beginning, bypassing the need for a http server. This is great. PSVue + EPG service is turning out to be the best addon for US users (or at least for me :) ), Thanks @snoopyjoe  and @eracknaphobia for all the work!!!

PS: I think the next fix is finding a way to refresh m3u/epg data without restarting the prv. It causes the stream to stop playback when EPG.xml is done being written(and in rare cases the pvr to hang). Not a huge deal, but still somewhat annoying for a seamless experience.

**Comment by eracknaphobia** [comment]
Finally got around to testing @snoopyjoe 's test version and I agree, this should be the way to go with the service. Much faster, less overhead and less to go wrong. We loose support for 17 and below but I think that's a sacrifice worth making for progress and stability.

@snoopyjoe would you want to do a pull request for the updates you've made?

**Comment by snoopyjoe** [comment]
I'll create a PR later today. I was still trying to figure out a way the playlist can work on v17.

**Comment by JohnMH** [comment]
Doesn't seem to be working even with plugin:// URIs now. I can't even play a stream directly using the plugin:// URI manually, for some reason. I'll either find or file an issue with the other plugin, and try to fix it there.

**Comment by JohnMH** [comment]
Updated the PS Vue plugin and opening URIs directly works again, just not from the PVR menu..
